### DISCUSSION TITLE
How to exclude node_modules from "Search project files" (CMD + P)?

### DISCUSSION THREAD

**Discussion by KatSick** [zed-industries/zed#6553] [Feature Requests]
Have you been wondering how to exclude node_modules from the "Search project files" function in Zed (CMD + P)?

**Comment by JosephTLyons** [comment]
There is no good way to do this at the moment.  That being said, this week, we should be landing the ability to include and exclude certain paths from the project search.  It will be a minimal viable feature, in that you type in it each time, but in the future, we will likely add some sort of project or global setting to tell zed what to always include / exclude.

**Reply by JosephTLyons** [reply]
Sorry, I misunderstood your original question.  This wasn't about project search, but the file finder.  We are starting work on project settings, which should enable us to add some sort of include / exclude settings for each project.

The feature to include / exclude a project search landed in v0.86.1.  This isn't a project-specific setting that you can turn on and forgot, but at least you can exclude them on each project search.

Also, I believe Zed will not collect files inside of directories that are `.gitignore`d, so if you are able to toss `node_modules` into the `.gitignore` file, it should filter those out for you.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
